src/components/screens/saved-routes-screen/saved-routes-screen.tsx

```diff
@@ -64,61 +65,44 @@ class SavedRoutes extends React.Component<Props, State> {
   /**
    * Component render method
    */
-  render() {
-    const { routes, redirect } = this.state;
+  public render() {
+    const { redirect, deleteDialogVisible, routeToDelete, error } = this.state;
     const { accessToken, keycloak, classes } = this.props;
-    const routeCards = routes.map(this.renderRouteCard);
- 
-    if (redirect) {
-      return (
-        <Redirect to="/map" push={ true }/>
-      );
-    }
 
     return (
-      <AppLayout accessToken={ accessToken } keycloak={ keycloak }>
+      <AppLayout accessToken={ accessToken } keycloak={ keycloak } error={ error } clearError={ this.clearError } redirectTo={ redirect }>
         <Container>
           <Typography className={ classes.title } variant="h3" component="h1">
             { strings.savedRoutes }
           </Typography>
           {
-            this.state.loadingRoutes &&
-            <CircularProgress className={ classes.primaryLoader } />
+            this.renderRouteCards()
           }
 
-          {
-            !this.state.loadingRoutes && routeCards 
-          }
-
-          { 
-            this.renderDeleteDialog() 
-          }
+        <DeleteDialogue routeDeleteInitiated={ deleteDialogVisible } routeToDelete={ routeToDelete } onDeleteConfirm={ this.deleteRoute } onDeleteCancel={ this.closeDeleteDialog } />
         </Container>
       </AppLayout>
     );
   }
 
   /**
-   * Updates the list for routes
+   * Renders route cards
+   * 
+   * @returns a rendered route cards
    */
-  private updateRoutesList = async () => {
-    const { accessToken } = this.props;
-
-    if (!accessToken) {
-      return;
-    }
-
-    this.setState({ loadingRoutes: true });
-
-    try {
-      const routesApi = Api.getRoutesApi(accessToken);
-      const routes = await routesApi.listRoutes();
-      this.setState({ routes });
-    } catch (error) {
-      console.log(error);
+  private renderRouteCards = () => {
+    const { classes } = this.props;
+    const { routes, loadingRoutes } = this.state;
+    const routeCards = routes.map(this.renderRouteCard);
+    if (!loadingRoutes) {
+      return (
+        routeCards
+      );
```

Comment:
React render methods should return only single element

```suggestion
    if (!loadingRoutes) {
      return (
        <>
          { routes.map(this.renderRouteCard) }
        </>
      );
```